example/offload-template/src/ActionInitialization.cc

```diff
@@ -0,0 +1,46 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file example/offload-template/src/ActionInitialization.cc
+//---------------------------------------------------------------------------//
+#include "ActionInitialization.hh"
+
+#include <accel/TrackingManagerOffload.hh>
+
+#include "Celeritas.hh"
+#include "EventAction.hh"
+#include "G4Electron.hh"
+#include "G4Gamma.hh"
+#include "G4Positron.hh"
+#include "PrimaryGeneratorAction.hh"
+#include "RunAction.hh"
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct empty.
+ */
+ActionInitialization::ActionInitialization() : G4VUserActionInitialization() {}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set up all user actions and Celeritas offload interface.
+ */
+void ActionInitialization::Build() const
+{
+    // Construct Celeritas offloading interface
+    CelerSimpleOffload().Build(
+        &CelerSetupOptions(), &CelerSharedParams(), &CelerLocalTransporter());
```

Comment:
Should this be in BuildForMaster  (assuming that the default run manager (tasking run manager) is used)?

Reply:
It probably should support MT? Most new Geant4 apps should.

Reply:
Yeah, I thought of that. But then we have to require a Geant4 install with `GEANT4_BUILD_MULTITHREADED=ON` for a tiny example. Fairly, that would be standard for most cases, but seemed simpler to run a single-threaded one, so any compilation works, and explain how MT should be handled in a comment.

Reply:
The `BuildForMaster` member function can be supplied always. If Geant4 is _not_ MT enabled, it'll never be called, but by being there both cases can be supported.